anesthetic/plotting/_matplotlib/core.py

```diff
@@ -27,7 +31,23 @@ def __init__(self, data, *args, **kwargs):
 
     def _get_index_name(self):
         if isinstance(self.data, _WeightedObject):
-            return np.squeeze(self.data.drop_weights().index.names)
+            if isinstance(self.data.drop_weights().index, ABCMultiIndex):
+                name = self.data.drop_weights().index.names
+                if com.any_not_none(*name):
+                    name = ",".join([pprint_thing(x) for x in name])
+                else:
+                    name = None
+            else:
+                name = self.data.drop_weights().index.name
+                if name is not None:
+                    name = pprint_thing(name)
+
+            # GH 45145, override the default axis label if one is provided.
+            index_name = self._get_custom_index_name()
+            if index_name is not None:
+                name = pprint_thing(index_name)
+
+            return name
```

Comment:
This should be covered by a test for the use-case by which you discovered you needed it.